src/simulation/TxGenerator.h

```diff
@@ -0,0 +1,160 @@
+
+    std::pair<TestAccountPtr, TransactionTestFramePtr>
+    createUploadWasmTransaction(uint32_t ledgerNum, uint64_t accountId,
+                                xdr::opaque_vec<> const& wasm,
+                                LedgerKey const& contractCodeLedgerKey,
+                                std::optional<uint32_t> maxGeneratedFeeRate);
+    std::pair<TestAccountPtr, TransactionTestFramePtr>
+    createContractTransaction(uint32_t ledgerNum, uint64_t accountId,
+                              LedgerKey const& codeKey,
+                              uint64_t contractOverheadBytes,
+                              uint256 const& salt,
+                              std::optional<uint32_t> maxGeneratedFeeRate);
+
+    std::pair<TestAccountPtr, TransactionTestFramePtr>
+    invokeSorobanLoadTransaction(uint32_t ledgerNum, uint64_t accountId,
+                                 TxGenerator::ContractInstance const& instance,
+                                 uint64_t contractOverheadBytes,
+                                 std::optional<uint32_t> maxGeneratedFeeRate);
+    std::pair<TestAccountPtr, TransactionTestFramePtr>
+    invokeSorobanCreateUpgradeTransaction(
+        uint32_t ledgerNum, uint64_t accountId, SCBytes const& upgradeBytes,
+        LedgerKey const& codeKey, LedgerKey const& instanceKey,
+        std::optional<uint32_t> maxGeneratedFeeRate);
+    std::pair<TestAccountPtr, TransactionTestFramePtr>
+    sorobanRandomWasmTransaction(uint32_t ledgerNum, uint64_t accountId,
+                                 uint32_t inclusionFee);
+
+    std::pair<TestAccountPtr, TransactionTestFramePtr>
+    pretendTransaction(uint32_t numAccounts, uint32_t offset,
+                       uint32_t ledgerNum, uint64_t sourceAccount,
+                       uint32_t opCount,
+                       std::optional<uint32_t> maxGeneratedFeeRate);
+
+    int generateFee(std::optional<uint32_t> maxGeneratedFeeRate, size_t opsCnt);
+
+    std::pair<TestAccountPtr, TestAccountPtr>
+    pickAccountPair(uint32_t numAccounts, uint32_t offset, uint32_t ledgerNum,
+                    uint64_t sourceAccountId);
+
+    ConfigUpgradeSetKey
+    getConfigUpgradeSetKey(SorobanUpgradeConfig const& upgradeCfg,
+                           Hash const& contractId) const;
+
+    SCBytes getConfigUpgradeSetFromLoadConfig(
+        SorobanUpgradeConfig const& upgradeCfg) const;
+
+    std::map<uint64_t, TestAccountPtr> const& getAccounts();
+
+    medida::Counter const& GetApplySorobanSuccess();
+    medida::Counter const& GetApplySorobanFailure();
+
+    void reset();
+
+  private:
+    std::pair<SorobanResources, uint32_t> sorobanRandomUploadResources();
+
+    // Calculates total size we'll need to read for all specified keys
+    uint64_t bytesToRead(xdr::xvector<stellar::LedgerKey> const& keys);
+
+    void updateMinBalance();
+
+    Application& mApp;
+
+    // Accounts cache
+    std::map<uint64_t, TestAccountPtr> mAccounts;
```

Comment:
This map is used by loadgen, so I left it as is in case it was intentionally chosen to be a map. There one function that loops over it. It would probably be fine to switch, but with loadgen being as finicky as it is, I'd rather not mess with it even more.